# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/wifi/nsWifiScannerWin.cpp
# Commit: afd62e82fe83
# Full Hash: afd62e82fe8320776e9696e3a8881eaae56b8822
# Author: David Parks <daparks@mozilla.com>
# Date: 2023-05-11 04:06:39
# Regressor Bug: 1810421
# File Overlap Count: 2
# Description:
#   Bug 1810421: Refactor wifi scanning to consolidate generic code r=necko-reviewers,kershaw,valentin
#   
#   Reorganizes the wifi-scanning code and makes more of it platform-generic
#   to ease the transition from polling the wifi to usually scanning only on
#   network changes.  This is essentially just moving files/code around and
# ==============================================================================

diff -r 9c94ab8b34c0 -r afd62e82fe83 netwerk/wifi/nsWifiScannerWin.cpp
--- a/netwerk/wifi/nsWifiScannerWin.cpp	Wed May 10 19:29:32 2023 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,64 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#include "nsWifiMonitor.h"
-
-// moz headers (alphabetical)
-#include "nsCOMArray.h"
-#include "nsComponentManagerUtils.h"
-#include "nsServiceManagerUtils.h"
-#include "nsWifiAccessPoint.h"
-#include "win_wifiScanner.h"
-
-using namespace mozilla;
-
-/**
- * `nsWifiMonitor` is declared in the cross-platform nsWifiMonitor.h and
- * is mostly defined in the cross-platform nsWifiMonitor.cpp. This function
- * is implemented in various platform-specific files but the implementation
- * is almost identical in each file. We relegate the Windows-specific
- * work to the `WinWifiScanner` class and deal with non-Windows-specific
- * issues like calling listeners here. Hopefully this file can be merged
- * with the other implementations of `nsWifiMonitor::DoScan` since a lot
- * of the code is identical
- */
-nsresult nsWifiMonitor::DoScan() {
-  if (!mWinWifiScanner) {
-    mWinWifiScanner = MakeUnique<WinWifiScanner>();
-    if (!mWinWifiScanner) {
-      // TODO: Probably return OOM error
-      return NS_ERROR_FAILURE;
-    }
-  }
-
-  // Regularly get the access point data.
-
-  nsCOMArray<nsWifiAccessPoint> lastAccessPoints;
-  nsCOMArray<nsWifiAccessPoint> accessPoints;
-
-  do {
-    accessPoints.Clear();
-    nsresult rv = mWinWifiScanner->GetAccessPointsFromWLAN(accessPoints);
-    if (NS_FAILED(rv)) {
-      return rv;
-    }
-
-    bool accessPointsChanged =
-        !AccessPointsEqual(accessPoints, lastAccessPoints);
-    ReplaceArray(lastAccessPoints, accessPoints);
-
-    rv = CallWifiListeners(lastAccessPoints, accessPointsChanged);
-    NS_ENSURE_SUCCESS(rv, rv);
-
-    // wait for some reasonable amount of time.  pref?
-    LOG(("waiting on monitor\n"));
-
-    ReentrantMonitorAutoEnter mon(mReentrantMonitor);
-    if (mKeepGoing) {
-      mon.Wait(PR_SecondsToInterval(kDefaultWifiScanInterval));
-    }
-  } while (mKeepGoing);
-
-  return NS_OK;
-}